src/core/services/interimElement/interimElementProvider.js

```diff
@@ -0,0 +1,278 @@
+angular
+  .module('material.core')
+  .provider('$$interimElement', InterimElementProvider);
+
+function InterimElementProvider() {
+  function test () {
+    console.log("LOL");
+  }
+
+  // This generates the basic interim element service for this provider.
+  InterimElementConstructor.$get = InterimElementService;
+
+  // This returns our actual provider factory which generates the interim element.
+  return InterimElementConstructor;
+}
+
+function InterimElementConstructor(interimName) {
+  var provider;
+  var defaultMethods = ['onHide', 'onShow', 'onRemove'];
+
+  var providerConfig = {
+    globalMethods: {},
+    presets: {}
+  };
+
+  _addPreset('build', {
+    methods: ['controller', 'controllerAs', 'resolve',
+      'template', 'templateUrl', 'themable', 'transformTemplate', 'parent']
+  });
+
+  return provider = {
+    setDefaults: _setDefaults,
+    addPreset: _addPreset,
+    addMethod: _addMethod,
+    $get: factory
+  };
+
+  function _setDefaults(definition) {
+    definition = parseDefinition(definition);
+
+    providerConfig.optionsFactory = definition.options;
+    providerConfig.methods = defaultMethods.concat(definition.methods || []);
+
+    return provider;
+  }
+
+  function _addPreset(name, definition) {
+    definition = parseDefinition(definition);
+
+    providerConfig.presets[name] = {
+      argOption: definition.argOption,
+      optionsFactory: definition.options,
+      methods: defaultMethods.concat(definition.methods)
+    };
+
+    return provider;
+  }
+
+  function _addMethod(name, fn) {
+    providerConfig.globalMethods[name] = fn;
+
+    return provider;
+  }
+
+  function parseDefinition(definition, name) {
+    definition.methods = definition.methods || [];
+    definition.options = definition.options || function() { return {}; };
+
+    if (name) {
+      if (/^cancel|hide|show$/.test(name)) {
+        throw new Error("Preset '" + name + "' in " + interimName + " is reserved!");
+      } else if (definition.methods.indexOf('_options') != -1) {
+        throw new Error("Method '_options' in " + interimName + " is reserved!");
+      }
+    }
+
+    return definition;
+  }
+
+  function factory($$interimElement, $injector) {
+    // This is needed, otherwise ng-annotator can't detect this function as a factory $get.
+    'ngInject';
```

Comment:
Yes, indeed, using that in the doc string causes warning like these:
![image](https://cloud.githubusercontent.com/assets/4987015/13999990/5b34fcec-f13f-11e5-9217-773a5686d587.png)